File: app/src/main/java/com/shetj/diyalbume/rx/LoginObservable.java
package com.shetj.diyalbume.rx;




import io.reactivex.Observable;
import io.reactivex.Observer;
import io.reactivex.android.MainThreadDisposable;

/**
 * <b>@packageName：</b> com.shetj.diyalbume.rx<br>
 * <b>@author：</b> shetj<br>
 * <b>@createTime：</b> 2019/3/15 0015<br>
 * <b>@company：</b><br>
 * <b>@email：</b> 375105540@qq.com<br>
 * <b>@describe</b><br>
 */
class LoginObservable extends Observable<Object> {
	private String name;
	private String password;
	public LoginObservable(String name, String passWord) {
		this.name = name;
		this.password = passWord;
	}

	@Override
	protected void subscribeActual(Observer<? super Object> observer) {
		observer.onSubscribe(new Listener(observer));
	}

	static final class Listener extends MainThreadDisposable   {
		private final Observer<? super Object> observer;

		Listener( Observer<? super Object> observer) {
			this.observer = observer;
		}


		@Override
		protected void onDispose() {
			observer.onComplete();
		}
	}
}

File: app/src/main/java/com/shetj/diyalbume/rx/LoginObservable.kt
package com.shetj.diyalbume.rx


import io.reactivex.observers.DefaultObserver

/**
 * **@packageName：** com.shetj.diyalbume.rx<br></br>
 * **@author：** shetj<br></br>
 * **@createTime：** 2019/3/15 0015<br></br>
 * **@company：**<br></br>
 * **@email：** 375105540@qq.com<br></br>
 * **@describe**<br></br>
 */
internal class LoginObservable(private val name: String, private val password: String) : DefaultObserver<Any>() {

    override fun onComplete() {

    }

    override fun onNext(t: Any) {

    }

    override fun onError(e: Throwable) {

    }

}
